mteb/models/model2vec_models.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Sequence
+from functools import partial
+from typing import Any
+
+import numpy as np
+
+from mteb.encoder_interface import PromptType
+from mteb.model_meta import ModelMeta
+
+from .wrapper import Wrapper
+
+logger = logging.getLogger(__name__)
+
+
+class Model2VecWrapper(Wrapper):
+    def __init__(
+        self,
+        model: str,
+        **kwargs,
+    ) -> None:
+        """Wrapper for Model2Vec models.
+
+        Args:
+            model: The Model2Vec model to use. Can be a string (model name), a SentenceTransformer model, or a CrossEncoder model.
+        """
+        self.model = model
+        self.static_model = StaticModel.from_pretrained(self.model)
+
+    def encode(
+        self,
+        sentences: Sequence[str],
+        **kwargs: Any,
+    ) -> np.ndarray:
+        """Encodes the given sentences using the encoder.
+
+        Args:
+            sentences: The sentences to encode.
+            **kwargs: Additional arguments to pass to the encoder.
+
+        Returns:
+            The encoded sentences.
+        """
+        return self.static_model.encode(sentences)
+
+
+m2v_base_glove_subword = ModelMeta(
+    loader=partial(
+        Model2VecWrapper,
+        model="minishlab/M2V_base_glove_subword",
+    ),
+    name="minishlab/M2V_base_glove_subword",
+    languages=["eng_Latn"],
+    open_weights=True,
+    revision="5f4f5ca159b7321a8b39739bba0794fa0debddf4",
+    release_date="2024-10-14",  # blog post date
+    n_parameters=103 * 1e6,
+    max_tokens=1000000,  # I took this from the current leaderboard, I'm not sure if it's unlimited in theory, tough
```

Comment:
It probably is - but there might be a point where the average isn't that meaningful

Reply:
Maybe np.inf?

Reply:
This is indeed set to a high default since there is technically no limit, but ideally this should be set to something like inf (if possible). I created an issue for this [here](https://github.com/embeddings-benchmark/leaderboard/issues/53) on the leaderboard repo as well, tbh I'm not sure what the best solution is. As Kenneth said, it is technically possible to pass in an unlimited number of tokens, but the mean of that probably isn't meaningful anymore.

Reply:
Let us do np.inf, then we will figure out how to best display it in the leaderboard